nibabel/cifti2/cifti2.py

```diff
@@ -1442,4 +1442,6 @@ def save(img, filename):
     filename : str
         filename to which to save image
     """
+    if img.nifti_header.get_intent()[0] == 'none':
+        raise AttributeError("CIFTI image has an invalid intent code.")
```

Comment:
Actually, I think I would rather set the intent by default to `'NIFTI_INTENT_CONNECTIVITY_UNKNOWN'`, and I think we should do this in `Cifti2Image.to_file_map()`, maybe after resetting `pixdim`.

```Python
if header.get_intent()[0] == 'none':
    header.set_intent('NIFTI_INTENT_CONNECTIVITY_UNKNOWN')
```

This feels a little cleaner to me, as the CIFTI equivalent of saving a NIFTI without setting an intent code, which we also allow.